thirdeye-server/src/main/java/ai/startree/thirdeye/rca/RcaInfoFetcher.java

```diff
@@ -145,6 +145,9 @@ public RcaInfo getRcaInfo(final AnomalyDTO anomalyDTO)
           .setName(metricName)
```

Comment:
I think we can skip fetching from the persistence layer altogether.
The long term plan is to use this metric abstraction less and less and eventually remove it. (it has other issues, like ThirdEye needs to maintain the metrics in sync with the table schema).

So I think we can just construct the metricConfigDTO manually likes it's done line 144. 
```
metricConfigDTO = new MetricConfigDTO()
           .setDataset(datasetName)
          .setName(metricName)
          .setDefaultAggFunction(
         optional(metadataMetricDTO.getDefaultAggregationFunction).orElse(
         optional(alertMetadataDto.getDetectionAggregationFunction()).orElse(COUNT));
```

Reply:
note this will change the logic to fetch the dataset below, we will need to fetch by name, namespace in all cases (this is fine)